### ISSUE TITLE
SC2094, SC2035, SC2034, SC2205, SC2283  ternary arithmetic false positives

### ISSUE THREAD

**Issue by apprehensions** [koalaman/shellcheck#2984]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2094, SC2035, SC2034, SC2205, SC2283  
- My shellcheck version (`shellcheck --version` or "online"): 0.10.0 & online
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

foo=1
bar=2

: "$((foo =< bar ? foo : bar))"
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
foo=1
^-- [SC2034](https://www.shellcheck.net/wiki/SC2034) (warning): foo appears unused. Verify use (or export if used externally).
 
bar=2
^-- [SC2034](https://www.shellcheck.net/wiki/SC2034) (warning): bar appears unused. Verify use (or export if used externally).
 
[Line 6:](javascript:setPosition(6, 4))
: "$((foo =< bar ? foo : bar))"
   ^-- [SC1102](https://www.shellcheck.net/wiki/SC1102) (error): Shells disambiguate $(( differently or not at all. For $(command substitution), add space after $( . For $((arithmetics)), fix parsing errors.
     ^-- [SC2205](https://www.shellcheck.net/wiki/SC2205) (warning): (..) is a subshell. Did you mean [ .. ], a test expression?
          ^-- [SC2283](https://www.shellcheck.net/wiki/SC2283) (error): Remove spaces around = to assign (or use [ ] to compare, or quote '=' if literal).
             ^-- [SC2094](https://www.shellcheck.net/wiki/SC2094) (info): Make sure not to read and write the same file in the same pipeline.
                 ^-- [SC2035](https://www.shellcheck.net/wiki/SC2035) (info): Use ./*glob* or -- *glob* so names with dashes won't become options.
                         ^-- [SC2094](https://www.shellcheck.net/wiki/SC2094) (info): Make sure not to read and write the same file in the same pipeline.

$
```

#### Here's what I wanted or expected to see:

```
$ shellcheck myscript
No issues detected!

$ 
```

**Comment by e28eta** [comment]
Is this possibly caused by a typo in your comparison operator? '=<' looks wrong to me, I think you want '<=' instead

**Comment by apprehensions** [comment]
Shellcheck doesn't know that, and won't report it, leaving me confused!

Yes, you are correct, my bad.
